src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/RecalcEngineTests.cs

```diff
@@ -711,6 +711,39 @@ public void ImperativeUserDefinedFunctionTest(string udfExpression, string expre
             }
         }
 
+        [Theory]
+
+        [InlineData(
+            "MismatchType():Number = { 1+3; Color.Blue; };",
+            true,
+            true,
+            36,
+            41)]
+        [InlineData(
+            "MatchType():Text = { 4; 3 };",
+            false,
+            true,
+            0,
+            0)]
+        public void TestMismatchReturnType(string udfExpression, bool expectedError, bool allowSideEffects, int min, int lim)
```

Comment:
>allowSideEffects [](http://example.com/codeflow?start=90&length=16)

For behavior UDFs we always need the allowSideEffects to be true, so there's no need to pass this along